src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_HEARING_WINDOW;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_ORG_INDIVIDUAL_DETAILS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_PANEL_REQUIREMENTS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_RELATED_PARTY_DETAILS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_REQUEST_DETAILS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_UNAVAILABILITY_DOW_DETAILS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_UNAVAILABILITY_RANGES_DETAILS;
+
+@Service
+@Slf4j
+public class HearingManagementServiceImpl implements HearingManagementService {
+
+    @Autowired
+    public HearingManagementServiceImpl() {
+        //Do nothing
+    }
+
+    @Override
+    public void validateHearingRequest(HearingRequest hearingRequest) {
+        validateHearingRequestDetails(hearingRequest);
+        validateRequestDetails(hearingRequest.getRequestDetails());
+        validateHearingDetails(hearingRequest.getHearingDetails());
+        validateCaseDetails(hearingRequest.getCaseDetails());
+        if (hearingRequest.getPartyDetails() != null) {
+            validatePartyDetails(hearingRequest.getPartyDetails());
+        }
+
+    }
+
+    private void validatePartyDetails(PartyDetails[] partyDetails) {
+        for (PartyDetails partyDetail : partyDetails) {
+            if ((partyDetail.getIndividualDetails() != null && partyDetail.getOrganisationDetails() != null)
+                || (partyDetail.getIndividualDetails() == null && partyDetail.getOrganisationDetails() == null)) {
+                throw new BadRequestException(INVALID_ORG_INDIVIDUAL_DETAILS);
+            }
+            if (partyDetail.getUnavailabilityDow() != null && partyDetail.getUnavailabilityDow().length == 0) {
+                throw new BadRequestException(INVALID_UNAVAILABILITY_DOW_DETAILS);
+            }
+            if (partyDetail.getUnavailabilityRanges() != null && partyDetail.getUnavailabilityRanges().length == 0) {
+                throw new BadRequestException(INVALID_UNAVAILABILITY_RANGES_DETAILS);
+            }
+            if (partyDetail.getIndividualDetails().getRelatedParties() != null
+                && partyDetail.getIndividualDetails().getRelatedParties().length == 0) {
+                throw new BadRequestException(INVALID_RELATED_PARTY_DETAILS);
+            }
+        }
+    }
+
+    private void validateHearingRequestDetails(HearingRequest hearingRequest) {
+        if (hearingRequest.getRequestDetails() == null
+            && hearingRequest.getHearingDetails() == null
+            && hearingRequest.getCaseDetails() == null) {
+            throw new BadRequestException(INVALID_HEARING_REQUEST_DETAILS);
+        }
+    }
+
+    private void validateRequestDetails(RequestDetails requestDetails) {
+        if (requestDetails == null) {
```

Comment:
For these checks where they're only checking `== null`, can't we just use `@NotNull` or similar instead?